clients/unshaded/src/main/java/tachyon/client/file/policy/LocalFirstPolicyFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file.policy;
+
+import java.util.List;
+
+import tachyon.client.block.BlockWorkerInfo;
+
+/**
+ * Factory that constructs a {@link LocalFirstPolicy}.
+ */
+public class LocalFirstPolicyFactory implements FileWriteLocationPolicyFactory<LocalFirstPolicy> {
```

Comment:
All `FooPolicyFactory` has the same pattern of code. Is it possible to make `FileWriteLocationPolicyFactory` a generic class (rather than a generic interface), so we don't need to repeatedly define the same subclass factories?